programs/create-genesis-ee/genesis_ee_builder.cpp

```diff
@@ -0,0 +1,186 @@
+#include "genesis_ee_builder.hpp"
+#include "golos_operations.hpp"
+#include <cyberway/genesis/genesis_ee_container.hpp>
+
+#define MEGABYTE 1024*1024
+
+#define MAP_FILE_SIZE uint64_t(1.5*1024)*MEGABYTE
+
+namespace cyberway { namespace genesis {
+
+genesis_ee_builder::genesis_ee_builder(const std::string& shared_file)
+        : maps_(shared_file, chainbase::database::read_write, MAP_FILE_SIZE) {
+    maps_.add_index<comment_header_index>();
+}
+
+genesis_ee_builder::~genesis_ee_builder() {
+}
+
+golos_dump_header genesis_ee_builder::read_header(bfs::fstream& in, uint32_t expected_version) {
+    golos_dump_header h{"", 0, 0};
+    in.read((char*)&h, sizeof(h));
+    if (in) {
+        EOS_ASSERT(std::string(h.magic) == golos_dump_header::expected_magic && h.version == expected_version, genesis_exception,
+            "Unknown format of the operation dump file.");
+    }
+    return h;
+}
+
+bool genesis_ee_builder::is_newer(const golos_dump_header& h, const comment_header& comment) {
+    return h.block_num > comment.block_num || (h.block_num == comment.block_num && h.op_in_block > comment.op_in_block);
+}
+
+void genesis_ee_builder::process_comments() {
+    std::cout << "-> Reading comments..." << std::endl;
+
+    bfs::fstream in(in_dump_dir_ / "comments");
+    while (in) {
+        auto h = read_header(in, 1);
+
+        if (!in) {
+            break;
+        }
+
+        uint64_t hash = 0;
+        fc::raw::unpack(in, hash);
+
+        auto comment_offset = uint64_t(in.tellg());
+
+        cyberway::golos::comment_operation cop;
+        fc::raw::unpack(in, cop);
+
+        const auto& comment_index = maps_.get_index<comment_header_index, by_hash>();
+        auto comment_itr = comment_index.find(hash);
+        if (comment_itr != comment_index.end()) {
+            maps_.remove(*comment_itr);
```

Comment:
Why remove?
May be simple changing of record is better?

Reply:
@afalaleev id should be new, for correct order.

Reply:
@afalaleev Hm, yes, comment can be before post:
Block 1. Post created
Block 2. Comment created
Block 3. Post edited
Is it wrong?